### PR TITLE
Writing_Linter_Bears.rst: Fix method definition

### PR DESCRIPTION
**Pull request by z1ggy-o**
Adds missing code and changes it from
create_arguments(config_file) --> create_arguments(self, config_file)

Fixes https://github.com/coala/coala/issues/5354

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by palash25** [comment]
LGTM :+1:

**Comment by jayvdb** [comment]
unack https://github.com/coala/coala/pull/5450/commits/fae55c6864a18924ef794923ecce7322fc3b36e7

**Comment by jayvdb** [comment]
See issue.

**Comment by z1ggy-o** [comment]
@ishanSrt I've read the comments in [#5354](https://github.com/coala/coala/issues/5354#issuecomment-387588240)
but not sure how to make the change.

Should I remove the `@staticmethod` and change `pass` to `return []` ?
like
```python
class SomeToolBear:
    def create_arguments(self, config_file):
        return []
```

**Comment by palash25** [comment]
>  using @staticmethod without self like CPPCheckBear does 

@ishanSrt its actually the other way around. CPPCheckBear doesn't use `@staticmethod` but uses `self`

File: docs/Developers/Writing_Linter_Bears.rst [hunk 1]
```diff
@@ -496,17 +496,40 @@ give us one, by passing the parameter ``global_bear=True``:
...
+    class PythonTestBear:
+        def create_arguments(self, config_file):
+            return '--lint', self.file_dict.keys()
+
+You can access the complete list of files using ``self.file_dict`` which return
+a dictionary of ``{filename: file contents}``.
+What you need to pay attention to is that putting the complete list of files on
+the command line will cause breakages when the length of the command line
+exceeds the OS permitted length, or the complete list of files is greater than
+the OS permitted number of args.
```
Comment (ishanSrt) [inline docs/Developers/Writing_Linter_Bears.rst]: break this long sentence
~~What you need to pay attention to is that putting the~~ --> Pay attention that putting the

**Comment by jayvdb** [comment]
gitmate-bot ff

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**